contracts/protocol/bases/FundsBase.sol

```diff
@@ -272,7 +272,7 @@ abstract contract FundsBase is Context {
                     );
                 } else {
                     if (returnAmount > 0) {
-                        IERC20(exchangeToken).safeApprove(exchange.mutualizerAddress, returnAmount);
+                        IERC20(exchangeToken).forceApprove(exchange.mutualizerAddress, returnAmount);
```

Comment:
```suggestion
                        IERC20(exchangeToken).safeIncreaseAllowance(exchange.mutualizerAddress, returnAmount);
```
?
This would allow the mutualizer contract to pull the funds asynchronously (maybe a stupid implementation, but can we avoid it?)